cmd/bnsd/x/username/username_test.go

```diff
@@ -30,27 +30,15 @@ func TestUsername(t *testing.T) {
 			WantDomain: "iov",
 		},
 		"too long name": {
-			Raw:        strings.Repeat("x", 65) + "*" + strings.Repeat("x", 6),
+			Raw:        strings.Repeat("x", 65) + "*iov",
 			WantName:   strings.Repeat("x", 65),
-			WantDomain: strings.Repeat("x", 6),
-			WantErr:    errors.ErrInput,
-		},
-		"too long domain": {
-			Raw:        strings.Repeat("x", 8) + "*" + strings.Repeat("x", 17),
-			WantName:   strings.Repeat("x", 8),
-			WantDomain: strings.Repeat("x", 17),
-			WantErr:    errors.ErrInput,
-		},
-		"too short name": {
-			Raw:        strings.Repeat("x", 3) + "*" + strings.Repeat("x", 6),
-			WantName:   strings.Repeat("x", 3),
-			WantDomain: strings.Repeat("x", 6),
+			WantDomain: "iov",
 			WantErr:    errors.ErrInput,
 		},
-		"too short domain": {
-			Raw:        strings.Repeat("x", 8) + "*" + strings.Repeat("x", 2),
-			WantName:   strings.Repeat("x", 8),
-			WantDomain: strings.Repeat("x", 2),
+		"all valid characters in name": {
+			Raw:        `abcdefghijklmnopqrstuvwxyz 0123456789.-_*iov`,
```

Comment:
q: why does this pass? space is not whitelisted.

Reply:
Good point and a great catch!